### ISSUE TITLE
fix: the `if:` condition model disagrees with the live binary in both directions

### ISSUE THREAD

**Issue by zernie** [zernie/agent-lint#214]
`src/core/hook-condition.ts` documents its own invariant in the module header:

> uncertainty can only ever ADD a run, never subtract one

Both directions are violated. Reproduced by running `claudeCodeHookProtocol.condition.matches(...)` out of a built `dist` against Claude Code **2.1.263**, comparing our verdict to whether the binary actually spawned the hook.

## 1. File patterns: we say "not run", CC runs it

```
Write(src/*)   file_path=/tmp/vigiles-harness-QcZKM2/src/x.ts   vigiles.runs=false   CC: FIRED
Write(.env*)   file_path=/tmp/p/.env.local                      vigiles.runs=false   CC: FIRED
```

This is the forbidden direction. `runs=false` renders as `⊘ not run — Claude Code would not run this hook`, so a guard that is working and firing is reported as unreachable.

It is not an edge case. Claude Code's contract passes an **absolute** `file_path`, while the idiomatic way to write the registration is a **relative** pattern (`Write(.env*)`). Every hook written that way is currently mis-reported.

## 2. Absolute program head: we say "runs", CC does not spawn

```
Bash(git push *--force*)   "/usr/bin/git push --force origin main"   vigiles.runs=true   CC: not spawned
Bash(git push *-f*)        "/usr/bin/git push --force origin main"   vigiles.runs=true   CC: not spawned
Bash(git push *)           "/usr/bin/git push --force origin main"   vigiles.runs=true   CC: not spawned
```

The allowed direction (over-reporting), but it produces a false accusation: we credit a hook with covering an input the binary never hands it. Any coverage number computed over absolute-head commands is too high.

## 3. The schema comment is stale

`hook-condition.ts` states the binary carries the `if` description string verbatim. On 2.1.263 the hook entry has **no `if` field in the zod schema** and the quoted description string is not in the bundle — while the feature itself still works. The comment should say what was actually verified and against which version, or stop claiming to quote the binary.

## Why this blocks other work

Anything that measures hook reachability — a coverage column, an audit ring, a survey of a corpus — currently runs through this model. A measurement taken with it inherits both errors, so this should land before the next one, not after.

## Fix shape

Both cases are about how the candidate path/command is normalized before matching, not about the matcher itself: case 1 needs the relative pattern resolved against the same root the absolute `file_path` is expressed in; case 2 needs the program head handled the way the binary handles it. Each needs a test in both directions — fires on a planted input, silent on a clean one — since a one-directional test would have passed throughout.
